### PR TITLE
Add project insights with JSON logging, chronological ordering, and ranking logic

### PR DESCRIPTION
**Pull request by Puneet-Maan**
## 📝 Description

Adds the initial Project Insights feature.

### Summary
- Introduces project_insights.py:
  - Records project insights to a JSON log.
  - Normalizes contributor data and computes stats.
  - Handles corrupted JSON files by stashing backups.
  - Provides:
    - record_project_insight
    - list_project_insights (chronological ordering)
    - rank_projects_by_contribution (global + per-contributor)
- Adds test suite test_project_insights.py covering:
  - Recording & listing insights  
  - Timestamp-based ordering  
  - Contribution ranking  
  - Corrupted file recovery behavior  

**Closes:** #113 #115 #117 

---

## 🔧 Type of Change

- [ ] 🐛 Bug fix  
- [x] ✨ New feature  
- [ ] 💥 Breaking change  
- [ ] 📚 Documentation added/updated  
- [x] ✅ Test added/updated  
- [ ] ♻️ Refactoring  
- [ ] ⚡ Performance improvement  

---

### Automated
- python3 -m unittest discover -s test -p "test_project_insights.py" -v
- All tests in test_project_insights.py pass:
  - Recording & listing insights  
  - Chronological order  
  - Ranking (global & per-contributor)  
  - top_n edge cases  
  - Corrupted storage handling  

### Manual
- Verified recording an insight creates the JSON log.
- Confirmed contributor normalization and stats fields.
- Verified corrupted JSON files get backed up automatically.

---

## ✓ Checklist

- [x] 🤖 GenAI was used in generating the code and I have performed a self-review  
- [x] 💬 Code is commented where needed  
- [x] 📖 Documentation updated if needed  
- [x] ⚠️ No new warnings introduced  
- [x] ✅ Tests added and passing locally  
- [x] 🔗 No missing dependent changes  
- [x] 📱 No UI changes (N/A)  

---

## 📸 Screenshots

image

### CODE REVIEW

**Review by mahigangal — APPROVED** [review]
Good job on finishing 3 features! Tests behave as expected, cover edge cases and all pass. Consider removing print statements in the tests which are fine for debugging but may create clutter, instead you can try using logging. Thanks for the detailed docstrings and helper methods.

File: src/project_insights.py [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+PathLike = Union[str, Path]
+
+DEFAULT_STORAGE = Path("User_config_files/project_insights.json")
+
+
+def _now_iso(ts: Optional[datetime] = None) -> str:
+    """Return an ISO 8601 timestamp in UTC."""
+    ts = ts or datetime.now(tz=timezone.utc)
+    return ts.astimezone(timezone.utc).isoformat()
+
```
Comment (ThunderIW) [inline src/project_insights.py]:
```suggestion
def _now_iso(ts: Optional[datetime] = None) -> str:
    """Return an ISO 8601 timestamp in UTC."""
    if ts is None:
        ts = datetime.now(timezone.utc)

    # Ensure timestamp is timezone-aware and in UTC
    if ts.tzinfo is None:
        ts = ts.replace(tzinfo=timezone.utc)

    return ts.astimezone(timezone.utc).isoformat()

```
This addition helps in making sure that your system handles naive datetimes, and forces them to be in UTC safely, something for you modified for next time

File: src/project_insights.py [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+    analyzed_at: str
+    languages: List[str] = field(default_factory=list)
+    frameworks: List[str] = field(default_factory=list)
+    skills: List[str] = field(default_factory=list)
+    project_type: str = "unknown"
+    detection_mode: str = "local"
+    duration_estimate: str = "unavailable"
+    hierarchy: JsonEntry = field(default_factory=dict)
+    contributors: ContributorData = field(default_factory=dict)
+    stats: JsonEntry = field(default_factory=dict)
```
Comment (ThunderIW) [inline src/project_insights.py]: Good use of the dataclass here, make it so that when the dict is saved, it's easy to retrieve info when we run through the dictionary by doing

**Review by ThunderIW — APPROVED** [review]
Overall, good work on this. I see no code-breaking issue. PR is approved and  look at my comments
